File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main(void){
    long long x = get_long_long("number: ");//prompts user for number
    long long copy = x;
    long long numDigits = 0;
    //do while loop that gets the length of the cc#
    do{
        ++numDigits;
        copy = copy / 10;
    } while ( copy );
    long long digits[numDigits];//creates array
    //while loop below puts each digit of the number in an array starting from the 1's place digit
    while( x >= 10 ){
        for(long long i=0; i<numDigits; i++){
            digits[i] = x % 10;
            x = x / 10;
        }
    }
    //for loop below prints item in the array

    //prints the length of the card number
    long long sumtwos[numDigits/2];
    long long k = 1;

    //while loop that puts the addition of each digit of everyother number*2 starting with second to last to an array
        for(long long sumtwosarray = 0; sumtwosarray<numDigits/2; sumtwosarray++){
            if(k< numDigits){
            int twos = 2*digits[k];
            if(twos>=10){
                sumtwos[sumtwosarray] = twos%10 + 1;
            }
            else {
                sumtwos[sumtwosarray] = twos;
            }
            k = k+2;
            }
        }
    long long fintwos = 0;
    //for loop that adds last array
    for( long long count = 0; count<numDigits/2; count++){
        fintwos = fintwos + sumtwos[count];
    }

    long long finregs = 0;
    //for loop that adds everyother number starting with last
    for(long long ones = 0; ones<=numDigits-1; ones= ones+2){
        finregs = finregs + digits[ones];
    }

    if(numDigits == 16){
        if((fintwos+finregs)%10 == 0){
            if(digits[numDigits-1]*10 + digits[numDigits - 2] == 51 || digits[numDigits-1]*10 + digits[numDigits - 2] == 52 || digits[numDigits-1]*10 + digits[numDigits - 2] == 53 || digits[numDigits-1]*10 + digits[numDigits - 2] == 54 || digits[numDigits-1]*10 + digits[numDigits - 2] == 55){
                printf("MASTERCARD\n");}
            else if(digits[numDigits-1] == 4){
                printf("VISA\n");
            }
            else printf("INVALID\n");
        }
        else printf("INVALID\n");
    }
    else if(numDigits == 13){
        if((fintwos+finregs)%10 == 0){
            if(digits[numDigits-1] == 4){
                printf("VISA\n");
            }
            else printf("INVALID\n");
        }
        else printf("INVALID\n");
    }
    else if(numDigits == 15){
        if((fintwos+finregs)%10 == 0){
            if(digits[numDigits-1]*10 + digits[numDigits - 2] == 34 || digits[numDigits-1]*10 + digits[numDigits - 2] == 37){
                printf("AMEX\n");
            }
            else printf("INVALID\n");
        }
        else printf("INVALID\n");
    }
    else printf("INVALID\n");
}

File: pset6/credit/credit.py
import cs50

cardnum = cs50.get_int("Number: ")
while cardnum <= 0:
    cardnum = cs50.get_int("Number: ")

# length = len(str(cardnum))

def cardvalid(card):
    length = len(str(card))
    strcard = str(card)
    if length != 15 and length != 16 and length != 13:
        print("INVALID")
        return 1
    total = 0
    count = 1
    while card > 0:
        if count % 2 == 1:
            total = total + card % 10
        else:
            total = total + (2 * (card % 10)) % 10 + (2 * (card % 10))//10
        count += 1
        card = card//10
    if total % 10 == 0:
        if length == 15 and strcard[0] == "3" and (strcard[1] == "4" or strcard[1] == "7"):
            print("AMEX")
            return 0
        elif (length == 13 or length == 16) and strcard[0] == "4":
            print("VISA")
            return 0
        elif length == 16 and (strcard[0] == "5" and (strcard[1] == "1" or strcard[1] == "2"
        or strcard[1] == "3" or strcard[1] == "4" or strcard[1] == "5")):
            print("MASTERCARD")
            return 0
        else:
            print("INVALID")
            return 0
    else:
        print("INVALID")
        return 1

cardvalid(cardnum)